### PR TITLE
move zxing to framework

### PR DESCRIPTION
**Pull request by michaelkirk**
and update build instructions

PTAL @charlesmchen 

If it looks good, I'll ff-merge the corresponding Carthage submodule commit.

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -1,55 +1,85 @@
...
+2.1) [CocoaPods](http://cocoapods.org) is used to manage the dependencies in our static library. Pods are setup easily and are distributed via a ruby gem. Follow the simple instructions on the website to setup. After setup, run the following command from the toplevel directory of Signal-iOS to download the dependencies for Signal-iOS:
 
 ```
 pod install
 ```
+
 If you are having build issues, first make sure your pods are up to date
+
 ```
 pod update
```
Comment (charlesmchen) [inline BUILDING.md]: I'm not sure we want to recommend a `pod update` when people have build issues.  Shouldn't it just be `pod install`?

File: BUILDING.md [hunk 2]
```diff
@@ -1,55 +1,85 @@
...
-5) In the Signal target on the General tab, change the Team drop down to your own. On the Capabilities tab turn off Push Notifications and Data Protection. Only Background Modes should remain on.
+In the Signal target on the General tab, change the Team drop down to
+your own. On the Capabilities tab turn off Push Notifications and Data
+Protection. Only Background Modes should remain on.
 
-6) Some of our build scripts, like running tests, expect your Derived
+Some of our build scripts, like running tests, expect your Derived
 Data directory to be `$(PROJECT_DIR)/build`. In Xcode, go to `Preferences-> Locations`,
 and set the "Derived Data" dropdown to "Relative" and the text field
 value to "build".
```
Comment (charlesmchen) [inline BUILDING.md]: Huh, I've never done this and tests run fine for me.  Why are we using a relative dir?

Comment (michaelkirk) [inline BUILDING.md]: > Why are we using a relative dir?

No reason that I know of, I just documented that we were. (It's hardcoded in some of the build settings). I'll double check that I can build without it and update the docs if so.

File: BUILDING.md (hunk 1 above)
Comment (michaelkirk) [inline BUILDING.md]: `pod repo update` is what we want I think. I'll update.

**Pull request merged**